go/runtime/txpool/txpool.go

```diff
@@ -303,9 +359,10 @@ func (t *txPool) GetPrioritizedBatch(offset *hash.Hash, limit uint32) []*transac
 
 func (t *txPool) GetKnownBatch(batch []hash.Hash) ([]*transaction.CheckedTransaction, map[hash.Hash]int) {
 	t.schedulerLock.Lock()
-	defer t.schedulerLock.Unlock()
+	schedulerQueue := t.schedulerQueue
```

Comment:
This is spooky, I take it this is to avoid adding `GetKnownBatchLocked`?

Reply:
Fixed.